frame/offences/src/lib.rs

```diff
@@ -103,6 +119,37 @@ pub mod pallet {
 	}
 }
 
+impl<T: Config> Pallet<T> {
+	fn max_session_report_age() -> SessionIndex {
```

Comment:
I think you actually need an interface that tells you this function. The decision of when an offence can be pruned is something that is up to the staking pallet, depending on if/how-much it decides to defer slashes. Given this, I would formulate something like: 


```rust
trait IsSessionOffencesProcessed {
  fn is_session_offences_processed -> bool {} 
}
```

and let this be implemented by `pallet_staking::Pallet`, and given to this pallet have it as input `type`.

Now, having said this out loud, it is reasonable to assume that whoever is `OnOffenceHandler` should have this knowledge. So `is_session_offences_processed` (or please choose a better name :D) should just be part of `OnOffenceHandler`. 

Would be good if @andresilva also verifies my thinking here, but this is what makes sense to me.